domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/frequencies/Frequency.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    @NotNull
+    public Integer getHeadwaySecs() {
+        return headwaySecs;
+    }
+
+    @Nullable
+    public ExactTimes isExactTimes() {
+        return exactTimes;
+    }
+
+    /**
+     * Builder class to create {@link Frequency} objects. Allows an unordered definition of the different attributes of
+     * {@link Frequency}.
+     */
+    public static class FrequencyBuilder {
+        private String tripId;
+        private Integer startTime;
+        private Integer endTime;
+        private Integer headwaySecs;
+        private ExactTimes exactTimes;
+        private String originalStartTime;
+        private String originalEndTime;
+        private Integer originalExactTimes;
+        private final List<Notice> noticeCollection = new ArrayList<>();
+
+
+        /**
+         * Sets field tripId value and returns this
+         *
+         * @param tripId identifies a trip class
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder tripId(@NotNull String tripId) {
+            this.tripId = tripId;
+            return this;
+        }
+
+        /**
+         * Sets field startTime value and returns this
+         *
+         * @param startTime identifies a start time for a trip service at the first stop
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder startTime(@NotNull String startTime) {
+            this.startTime = convertTimeToInteger(startTime);
+            this.originalStartTime = startTime;
+            return this;
+        }
+
+        /**
+         * Sets field endTime value and returns this
+         *
+         * @param endTime identifies an end time for a trip service at the first stop
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder endTime(@NotNull String endTime) {
+            this.endTime = convertTimeToInteger(endTime);
+            this.originalEndTime = endTime;
+            return this;
+        }
+
+        /**
+         * Sets field endTime value and returns this
+         *
+         * @param headwaySecs identifies times in seconds between departures in the same headway for a trip service
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder headwaySecs(@NotNull Integer headwaySecs) {
+            this.headwaySecs = headwaySecs;
+            return this;
+        }
+
+        /**
+         * Sets field exactTimes value and returns this
+         *
+         * @param exactTimes identifies an end time for a trip
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder exactTimes(@Nullable Integer exactTimes) {
+            this.exactTimes = ExactTimes.fromInt(exactTimes);
+            this.originalExactTimes = exactTimes;
+            return this;
+        }
+
+        public EntityBuildResult<?> build() {
+            noticeCollection.clear();
```

Comment:
Why is `noticeCollection` being cleared here? Doesn't seem like it's accessed anywhere besides this method.

Reply:
I seem to remember it's because with use a single instance of the Builder class to build all entities.

Reply:
Ah, yes - then it makes sense. But there isn't a general `.clear()` method on the builder? The current design is a bit dangerous when used as a single instance as it's assuming you always call all of the builder fields when building the object - if you don't then you keep one of the previous values when building.

Reply:
@barbeau @fabrice-v I modified the builder regarding this in my latest commit. Let me know if it makes sense.

Reply:
@maximearmstrong That's what I had in mind 👍 , although two more comments about this in-line.

Reply:
Oh, wait, I thought this comment was referring to the "HH:MM:SS" conversion - let me look at clearing the notice collection.

Reply:
I don't have all the code in front of me (just looking at GitHub), but definitely make sure there is a unit test for the notices being returned, as this code may be an issue:

```java
                toReturn = new EntityBuildResult<>(noticeCollection);
                noticeCollection.clear();
```

If `noticeCollection` is being passed by reference, you're actually clearing the list inside the `toReturn` instance with the call to `noticeCollection.clear()` - so it will be empty when returned.

I think the original code with `noticeCollection.clear();` may be clearer to understand if you just add a comment above, like:

```java
// Notices are generated for each entity build, so clear the list in case this Builder instance is being re-used
noticeCollection.clear();
```

Reply:
a general `.clear()` method has been implemented in https://github.com/MobilityData/gtfs-validator/pull/293